client/components/BreadCrumb/BreadCrumb.tsx

```diff
@@ -11,22 +11,19 @@ interface IMatches {
 }
 
 type HandleType = {
-  crumb: (param?: string) => React.ReactNode;
+  crumb: React.ReactNode;
 };
 
 function Breadcrumbs() {
   const matches: IMatches[] = useMatches();
 
-  const crumbs = matches
-    .filter((match) =>
-      Boolean(match.handle && (match.handle as HandleType).crumb),
-    )
-    .map((match) => {
-      const crumb = (match.handle as HandleType).crumb(
-        match.data as string | undefined,
-      );
-      return crumb as React.ReactNode;
-    });
+  const crumbs = useMemo(() => {
+    return matches
+      .filter((match) =>
+        Boolean(match.handle && (match.handle as HandleType).crumb),
+      )
```

Comment:
I don't think we need the `Boolean` function if the expression can return truthy

```suggestion
      .filter(
        (match) => match.handle != null && (match.handle as HandleType).crumb,
      )
```
We may be able to remove our `IMatches` interface altogether though if we assert UIMatch annotations 

```TypeScript
  const matches = useMatches() as UIMatch<unknown, HandleType>[];

  const crumbs = useMemo(() => {
    return matches
      .filter((match) => match.handle != null && match.handle.crumb)
      .map((match) => match.handle.crumb);
  }, [matches]);
  ```

Reply:
Nice catch, I removed those the `Boolean` function and used the `UIMatch` type instead.